keps/sig-windows/20190714-windows-csi-support.md

```diff
@@ -290,7 +296,7 @@ CSI Node Plugins need to execute certain privileged operations at the host level
 
 Registration of CSI Node Plugins on a Kubernetes node is handled by the Kubelet plugin watcher using the fsnotify package. This component needs to convert paths detected by fsnotify to Windows paths in handleCreateEvent() and handleDeleteEvent() before the paths are passed to AddOrUpdatePlugin() RemovePlugin() routines in desiredStateOfTheWorld. A new utility function, NormalizeWindowsPath(), will be added in utils to handle this.
 
-Given `os.ModeSocket` is not set on a socket file's `os.FileMode` in Windows, a specific check for `os.ModeSocket` in handleCreateEvent() will need to be relaxed.
+Given `os.ModeSocket` is not set on a socket file's `os.FileMode` in Windows (due to golang [issue](https://github.com/golang/go/issues/33357)), a specific check for `os.ModeSocket` in handleCreateEvent() will need to be relaxed for Windows until the golang issue is addressed.
```

Comment:
is there an alternative check that can be done on windows to avoid spawning lots of events for normal files/dirs (like ensuring it is not a directory, and is not a regular file)? I know there were bugs with the plugin manager getting overwhelmed trying to process change events for non-socket files under that dir.

Reply:
The code fragment I am referring to that needs to be relaxed is: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go#L193-L196

I think the rest of the code around the above does get invoked regardless of file type - so I don't think the invocation of `handleCreateEvent` is being filtered based on file type. 

We can explore sending down a FSCTL to the file to check for the Windows reparse points (through go wrappers) to align with `fi.Mode()&os.ModeSocket == 0`. I can call that plan out here.